DemoInfo/DemoParser.cs

```diff
@@ -506,9 +527,142 @@ public void ParseHeader()
 
 			Header = header;
 
-
 			if (HeaderParsed != null)
 				HeaderParsed(this, new HeaderParsedEventArgs(Header));
+
+				if (header.PlaybackTime == 0)  //header is corrupted
+					FixHeader();
+		}
+
+		void FixHeader()
+		{
+			ExpressParseToEnd();
+			Header.PlaybackTicks = IngameTick;
+			Header.PlaybackFrames = CurrentTick - (TickAtStart + 1);
+			Header.PlaybackTime = TickInterval * IngameTick;
+			
+			CurrentTick = 0;
+
+			//the ol' switcheroo
+			//It's probably safest and easiest to
+			//swap out for a fresh bitstream.
+			Input.Seek(0, System.IO.SeekOrigin.Begin);
```

Comment:
Alright then. As I said, there just needs to be an event that fires when we parse the ServerInfo and then applications can handle it. Perhaps it might make sense to include a wrapper class in demoinfo that handles the restart (instead of just documenting what to do somewhere) but I don't really care about that - feel free create one if you think there's widespread need for it.